<a href="https://colab.research.google.com/github/jorigo/Challenge-Tienda-de-Juan-DataScience/blob/main/Challenge_tienda_juan.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### Importación de datos



In [42]:
import pandas as pd
import matplotlib.pyplot as plt


url = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_1%20.csv"
url2 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_2.csv"
url3 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_3.csv"
url4 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_4.csv"

tienda = pd.read_csv(url)
tienda2 = pd.read_csv(url2)
tienda3 = pd.read_csv(url3)
tienda4 = pd.read_csv(url4)

tienda.head()

,Producto,Categoría del Producto,Precio,Costo de envío,Fecha de Compra,Vendedor,Lugar de Compra,Calificación,Método de pago,Cantidad de cuotas,lat,lon
0,Asistente virtual,Electrónicos,164300.0,6900.0,16/01/2021,Pedro Gomez,Bogotá,4,Tarjeta de crédito,8,4.60971,-74.08175
1,Mesa de comedor,Muebles,192300.0,8400.0,18/05/2022,Beatriz Morales,Medellín,1,Tarjeta de crédito,4,6.25184,-75.56359
2,Juego de mesa,Juguetes,209600.0,15900.0,15/03/2021,Juan Fernandez,Cartagena,1,Tarjeta de crédito,1,10.39972,-75.51444
3,Microondas,Electrodomésticos,757500.0,41000.0,03/05/2022,Juan Fernandez,Cali,4,Nequi,1,3.43722,-76.52250
4,Silla de oficina,Muebles,335200.0,20200.0,07/11/2020,Maria Alfonso,Medellín,5,Nequi,1,6.25184,-75.56359


#1. Análisis de facturación



##**Ingreso Total por cada tienda**


### Descripcion:
En este primer análisis, debes calcular el ingreso total de cada tienda. Sumando los valores de la columna Precio de cada conjunto de datos de la tienda para estimar los ingresos.

In [43]:
precioTotal1 = tienda['Precio'].sum()
precioTotal1

np.float64(1150880400.0)

In [44]:
precioTotal2 = tienda2['Precio'].sum()
precioTotal2

np.float64(1116343500.0)

In [7]:
precioTotal3 = tienda3['Precio'].sum()
precioTotal3

np.float64(1098019600.0)

In [10]:
precioTotal4 = tienda4['Precio'].sum()
precioTotal4

np.float64(1038375700.0)

#### La tienda con mayor ingresos fue la **Tienda 1**, con un ingreso de **$ 1.150.880.400**

# 2. Ventas por categoría

## Descripcion:
En este debes calcular la cantidad de productos vendidos por categoría en cada tienda. La idea es agrupar los datos por categoría y contar el número de ventas de cada tipo, mostrando las categorías más populares de cada tienda.


In [24]:
Venta_categoria1 = tienda.groupby('Categoría del Producto')['Producto'].count().reset_index(name="Ventas")
Venta_categoria1

,Categoría del Producto,Ventas
0,Artículos para el hogar,171
1,Deportes y diversión,284
2,Electrodomésticos,312
3,Electrónicos,448
4,Instrumentos musicales,182
5,Juguetes,324
6,Libros,173
7,Muebles,465


In [23]:
Venta_categoria2 = tienda2.groupby('Categoría del Producto')['Producto'].count().reset_index(name="Ventas")
Venta_categoria2

,Categoría del Producto,Ventas
0,Artículos para el hogar,181
1,Deportes y diversión,275
2,Electrodomésticos,305
3,Electrónicos,422
4,Instrumentos musicales,224
5,Juguetes,313
6,Libros,197
7,Muebles,442


In [25]:
Venta_categoria3 = tienda3.groupby('Categoría del Producto')['Producto'].count().reset_index(name="Ventas")
Venta_categoria3

,Categoría del Producto,Ventas
0,Artículos para el hogar,177
1,Deportes y diversión,277
2,Electrodomésticos,278
3,Electrónicos,451
4,Instrumentos musicales,177
5,Juguetes,315
6,Libros,185
7,Muebles,499


In [26]:
Venta_categoria4 = tienda4.groupby('Categoría del Producto')['Producto'].count().reset_index(name="Ventas")
Venta_categoria4

,Categoría del Producto,Ventas
0,Artículos para el hogar,201
1,Deportes y diversión,277
2,Electrodomésticos,254
3,Electrónicos,451
4,Instrumentos musicales,170
5,Juguetes,338
6,Libros,187
7,Muebles,480


#### Las categorias mas vendidas en todas las tiendas fueron **Muebles** y **Electronicos**.

# 3. Calificación promedio de la tienda


## Descripcion:
En este paso, debes calcular las calificaciones promedio de los clientes para cada tienda. El objetivo es conocer la satisfacción del cliente con los productos vendidos.

In [27]:
promedio1 = tienda['Calificación'].mean()
print(f"Tienda 1: {promedio1:.2f}")


Tienda 1: 3.98


In [30]:
promedio2 = tienda2['Calificación'].mean()
print(f"Tienda 2: {promedio2:.2f}")

Tienda 2: 4.04


In [28]:
promedio3 = tienda3['Calificación'].mean()
print(f"Tienda 3: {promedio3:.2f}")

Tienda 3: 4.05


In [29]:
promedio4 = tienda4['Calificación'].mean()
print(f"Tienda 4: {promedio4:.2f}")

Tienda 4: 4.00


### La tienda con el mejor promedio de calificacion es, la **Tienda 3**, con un promedio de **4.05**

# 4. Productos más y menos vendidos

## Descripcion:
En este paso, debes identificar qué productos fueron los más vendidos y los menos vendidos en cada tienda. Visualiza los resultados para que quede claro qué productos destacaron en ventas en cada tienda.

In [31]:
#Primero cuento los productos por grupos
ventas = tienda.groupby('Producto')['Producto'].count().sort_values(ascending=False)

#luego calculo el maximo Producto mas vendido
mas_vendido = ventas.idxmax()
max_cant = ventas.max()

print(f"El producto mas vendido fue {mas_vendido} con {max_cant} ventas")

#luego calculo el minimo del Producto menos vendido
menos_vendido = ventas.idxmin()
min_cant = ventas.min()

print(f"El producto menos vendido fue {menos_vendido} con {min_cant} ventas")

El producto mas vendido fue Armario con 60 ventas
El producto menos vendido fue Celular ABXY con 33 ventas


In [32]:
#Primero cuento los productos por grupos
ventas = tienda2.groupby('Producto')['Producto'].count().sort_values(ascending=False)

#luego calculo el maximo Producto mas vendido
mas_vendido = ventas.idxmax()
max_cant = ventas.max()

print(f"El producto mas vendido fue {mas_vendido} con {max_cant} ventas")

#luego calculo el minimo del Producto menos vendido
menos_vendido = ventas.idxmin()
min_cant = ventas.min()

print(f"El producto menos vendido fue {menos_vendido} con {min_cant} ventas")

El producto mas vendido fue Iniciando en programación con 65 ventas
El producto menos vendido fue Juego de mesa con 32 ventas


In [33]:
#Primero cuento los productos por grupos
ventas = tienda3.groupby('Producto')['Producto'].count().sort_values(ascending=False)

#luego calculo el maximo Producto mas vendido
mas_vendido = ventas.idxmax()
max_cant = ventas.max()

print(f"El producto mas vendido fue {mas_vendido} con {max_cant} ventas")

#luego calculo el minimo del Producto menos vendido
menos_vendido = ventas.idxmin()
min_cant = ventas.min()

print(f"El producto menos vendido fue {menos_vendido} con {min_cant} ventas")

El producto mas vendido fue Kit de bancas con 57 ventas
El producto menos vendido fue Bloques de construcción con 35 ventas


In [34]:
#Primero cuento los productos por grupos
ventas = tienda4.groupby('Producto')['Producto'].count().sort_values(ascending=False)

#luego calculo el maximo Producto mas vendido
mas_vendido = ventas.idxmax()
max_cant = ventas.max()

print(f"El producto mas vendido fue {mas_vendido} con {max_cant} ventas")

#luego calculo el minimo del Producto menos vendido
menos_vendido = ventas.idxmin()
min_cant = ventas.min()

print(f"El producto menos vendido fue {menos_vendido} con {min_cant} ventas")

El producto mas vendido fue Cama box con 62 ventas
El producto menos vendido fue Guitarra eléctrica con 33 ventas


# 5. Envío promedio por tienda

### Descrpcion:
En este paso, debes calcular el costo de envío promedio para cada tienda. El objetivo es comprender cuánto se gasta, en promedio, en el envío de cada tienda.  Sabiendo que el cliente paga el envío.

In [38]:
costo_envio1 = tienda['Costo de envío'].mean()
print(f"la Tienda 1, tiene un costo de envio de: {costo_envio1:.2f}")

la Tienda 1, tiene un costo de envio de: 26018.61


In [39]:
costo_envio2 = tienda2['Costo de envío'].mean()
print(f"la Tienda 2, tiene un costo de envio de: {costo_envio2:.2f}")


la Tienda 2, tiene un costo de envio de: 25216.24


In [40]:
costo_envio3 = tienda3['Costo de envío'].mean()
print(f"la Tienda 3, tiene un costo de envio de: {costo_envio3:.2f}")


la Tienda 3, tiene un costo de envio de: 24805.68


In [41]:
costo_envio4 = tienda4['Costo de envío'].mean()
print(f"la Tienda 4, tiene un costo de envio de: {costo_envio4:.2f}")

la Tienda 4, tiene un costo de envio de: 23459.46
